### ISSUE TITLE
Bring back m3u8 stream file

### ISSUE THREAD

**Issue by Porkepix** [LISTEN-moe/feedback#26]
There used to have a .m3u8 standard link file on the homepage.
Sadly it disappeared and the link ceased to work. It would be nice to have it back.
Especially, sometimes the other way it streams can sometimes be wilful. The m3u8 often was more reliable.

**Comment by Pitu** [comment]
For the time being you can point your audio player to either `https://listen.moe/stream`, `https://listen.moe/fallback` or `https://listen.moe/opus`

**Comment by Porkepix** [comment]
So, I tested those, while all of them are working, I would like to for which one what's the supposed differences, and thought at first for example first and third were the same, but it isn't.
- First one: Shows title informations, as track time shows time of the current tracker over zero. Also show some strange error between every track, example:
```
Invalid audio PTS: 319.078458 -> -0.002902
A: 00:04:23 / 00:00:00 Cache:  3s+0KB
File tags:
 Artist: 内田彩
 Album: Blooming!
 Title: Like a Bird
A: 00:04:28 / 00:00:00 Cache:  0s+65KB
Invalid audio PTS: 268.630204 -> -0.002902
A: 00:04:03 / 00:00:00 Cache:  4s+0KB
```
- Second one: Shows metadatas, no Audio PTS error, same timing informations
- Third one: No metadata shown, no Audio PTS error, timing is a huge time over zero, eg.
```
Playing: http://listen.moe/opus
 (+) Audio --aid=1 (opus 2ch 48000Hz)
AO: [coreaudio] 48000Hz stereo 2ch floatp
A: 494:08:21 / 00:00:00 Cache:  5s+0KB
```

Then after understanding this, a couple of questions:
- Are website/browser addon/mobile application using those stream URLs? To explain, I sometimes experience micro-freezes during play and sometimes audio just stop (while play didn't) and tried to understand where does it come from. Audio stopping happen very often on an old computer of mine with a very tired HDD, but also happen sometimes on a brand new one without any issue. Tried to look at browser console, looks like it does things with JavaScript to get chunks.

- As there is a recent news talking about K-pop added to the website, I guess there are such URLs for it to?

- Finally, should those URLs go back to the website like it used to be in the past? Maybe for exemple in the "APPS" tab, even if it's not really one?

EDIT: For the Audio PTS issue, number after the arrow is 100% of the time the same.

**Comment by iCrawl** [comment]
`/stream` plays opus audio wrapped inside a vorbis container
`/opus` plays the raw opus audio
`/fallback` plays mp3 audio

Everything is using those URLs, website switches between `/stream` and `/opus` (Firefox) while mobile uses `/fallback` for a wide support range.
I can't really replicate any of the "stuttering" you are experiencing myself. JavaScript doesn't really have to do anything with it, it's a simple `<audio></audio>` element that does all the heavy lifting. The only thing we do with JavaScript is pausing the audio and reenabling it when you press pause and play.

KPOP urls are:
`https://listen.moe/kpop/stream`, `https://listen.moe/kpop/opus` and `https://listen.moe/kpop/fallback`

Maybe, maybe not. At the end of the day even supplying a m3u8 file would be pretty useless. Because all that really is, is like you would open a `.txt` file now, and put `https://listen.moe/kpop/stream` in it, save the file and rename it to `m3u8`. So at the end of the day it's not really needed.

**Comment by Porkepix** [comment]
Yeah I just said m3u8 because that's what used to be on the website previously. But those URLs are fine too, just that I don't know where those are visible besides this issue.

Can't reproduce the lines in the browser console as of now, gonna take a look if I can get them again.

Any idea for the Audio PTS messages at each track change on the main URL though?

**Comment by iCrawl** [comment]
This could be an issue in the underlying process of converting and getting the files ready for the stream.
We send silence if something takes a bit longer to not interrupt the broadcast of the stream and make users not "restart" their connection to the endpoint.

I'll have a look eventually. Does it impact something? Because if it's just some annoying presentation timestamp displaying issue, I most certainly won't put it high on my list.

**Comment by Porkepix** [comment]
Maybe this can cause micro-blanks between tracks, but honestly I have no idea. So let's say this is just something pretty annoying.
A little bit more annoying though is the time presentation (current time over zero for two of the streams, and a total time over zero for the third)

EDIT for the previous message, URLs used to be on the homepage a long time ago
